CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs

```diff
@@ -55,65 +62,105 @@ public static class UploadToQnAMaker
                         FileName = blobName,
                         FileUri = fileUri
                     };
-                    msg.Add(queueMessage);
-
+                    msgs.Values.Add(queueMessage);
                     outRecord.Data["status"] = "InQueue";
                     return outRecord;
                 });
-
+            // Add a list of <= 10 files into one message to be extracted as a batch. 
```

Comment:
If the <10 files is a hard perf requirement of what can be handled by the system, let's make sure that we enforce this in the code somewhere.

Reply:
added a check